src/redux/rockets/rocketsSlice.jsx

```diff
@@ -0,0 +1,62 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+
+const dataUrl = 'https://api.spacexdata.com/v4/rockets';
+export const getRockets = createAsyncThunk('rocket/getRockets', () => fetch(dataUrl)
+  .then((res) => res.json())
+  .catch((err) => (err)));
+
+const initialState = {
+  rockets: [],
+  isLoading: true,
+};
+const rocketSlice = createSlice({
+  name: 'rocket',
+  initialState,
+  reducers: {
+    reserve: (state, action) => {
+      let index = null;
+      for (let i = 0; i < state.rockets.length; i += 1) {
+        if (state.rockets[i].id === action.payload) {
+          index = i;
+        }
+      }
+      for (let i = 0; i < state.rockets.length; i += 1) {
+        if (i === index) {
+          const newState = state;
+          newState.rockets[i].reserved = true;
+        }
+      }
+    },
```

Comment:
- [ OPTIONAL] I would be better to use `map()` or `filter()` to update the state. In the example below, I have highlighted how you can make your code  cleaner and more efficient .

```
const rocketSlice = createSlice({
  name: 'rocket',
  initialState,
  reducers: {
    reserve: (state, action) => {
      const { payload: id } = action;
      state.rockets = state.rockets.map(rocket => {
        if (rocket.id !== id) {
          return rocket;
        }
        return { ...rocket, reserved: true };
      });
    },
    cancelReserve: (state, action) => {
      const { payload: id } = action;
      state.rockets = state.rockets.map(rocket => {
        if (rocket.id !== id) {
          return rocket;
        }
        return { ...rocket, reserved: false };
      });
    },
  },
});

```